core/src/main/java/org/apache/lucene/search/vectorhighlight/CustomFieldQuery.java

```diff
@@ -75,6 +82,10 @@ void flatten(Query sourceQuery, IndexReader reader, Collection<Query> flatQuerie
         } else if (sourceQuery instanceof ToParentBlockJoinQuery) {
             ToParentBlockJoinQuery blockJoinQuery = (ToParentBlockJoinQuery) sourceQuery;
             flatten(blockJoinQuery.getChildQuery(), reader, flatQueries, boost);
+        } else if (sourceQuery instanceof BoostingQuery) {
```

Comment:
I think this is an ok way to handle `BoostingQuery` given that we can't modify the boost on some terms if other terms are found but it doesn't model how it works really. I think it is worth leaving a note comparing how `BoostingQuery` actually works and how you've modeled it and why it is ok.